app/models/folio/instance.rb

```diff
@@ -14,6 +14,44 @@ def self.fetch(request, _live_lookup = true)
 
       Folio::Instance.from_dynamic(data)
     end
+
+    def self.filter_parent_bound_withs(parent_bound_withs, hrid)
+      return [] unless parent_bound_withs
+
+      parent_bound_withs.flatten.compact.each do |bound_with_record|
+        @filter_parent_bound_withs ||= bound_with_record.dig('instance', 'items').each.map do |item|
+          boundwithhrids = item['boundWithHoldingsPerItem'].filter { |bw| bw.dig('instance', 'hrid') == hrid }.flatten
+          update_item_fields(item, boundwithhrids.first) if boundwithhrids.present?
+        end
+      end
+      @filter_parent_bound_withs.compact
+    end
+
+    def self.update_item_fields(item, holdings_record)
+      item['effectiveCallNumberComponents']['callNumber'] = holdings_record['callNumber']
+      item['volume'] = ''
+      item['enumeration'] = ''
+      item
+    end
+
+    def self.get_items(json)
+      hrid = json.fetch('hrid', '')
+      parent_bound_withs = parent_bound_withs(json)
+      return json.fetch('items', []).map { |item| Folio::Item.from_hash(item) } unless parent_bound_withs.present? && hrid
+
+      parent_bound_withs = parent_bound_withs.pluck('boundWithItem')
+      items = filter_parent_bound_withs(parent_bound_withs, hrid)
+      items += json['items'] if json['items']
+      items.compact.map { |item| Folio::Item.from_hash(item) }
+    end
+
+    def self.parent_bound_withs(json)
+      json.fetch('holdingsRecords', [])&.filter { |holding| holding['boundWithItem'] }
+    end
+
+    def self.child_bound_withs(json)
+      json['items']&.pluck('boundWithHoldingsPerItem')&.flatten&.compact
```

Comment:
If you do `return unless json['items']`, then you don't need all of these safe navigation operators.